control/control.go

```
@@ -226,6 +262,62 @@ func New(cfg *Config) *pluginControl {
 	return c
 }
 
+func (p *pluginControl) HandleGomitEvent(e gomit.Event) {
+	switch v := e.Body.(type) {
+	case *control_event.LoadPluginEvent:
+		// while task is running we care about collectors being loaded
+		// only collectors can affect current workflow by changing metric catalog
+		if core.PluginType(v.Type) == core.CollectorPluginType {
+			for taskID, td := range p.taskView {
+				// get all new metrics and collector plugins
+				newMetrics, newPlugins, _ := p.getMetricsAndCollectors(td.requested, td.configTree)
```

Is swallowing this error a good idea?